### ISSUE TITLE
Full DOMA test pools 

### ISSUE THREAD

**Issue by ageorget** [dCache/dcache#5059]
Hi,

As discussed previously with Paul, I have some issue with dCache pools used by DOMA TPC tests.
The pools started to became 100% full according to dCache but when I checked the real space used on the server (du -sh), I get only 50% space used. 

> --- rep (Repository manager) ---
> State : OPEN
> Files : 308
> Disk space
>     Total    : 2000G
>     Used     : 2147023165695    [0.9997855712170712]
>     Free     : 460482305    Gap : 4294967296
>     Precious : 0    [0.0]
>     Removable: 0    [0.0]

After restarting the pool, the total size of pinned files fell from 100% to 50% : 

> --- rep (Repository manager) ---
> State : OPEN
> Files : 308
> Disk space
>     Total    : 2000G
>     Used     : 1094186107702    [0.5095201114667579]
>     Free     : 1053297540298    Gap : 4294967296
>     Precious : 0    [0.0]
>     Removable: 0    [0.0]

No errors during pool initialization, no movers remained on the pool before restart, I don't see any error msg related with cleaner.
But on SpaceManager logs, I got these messages about TPC webdav failed transfers on DOMA pools : 
```
02 Sep 2019 14:54:04 (SpaceManager) [door:webdav-ccdcatli302@webdav-ccdcatli302Domain:AAWRkMuH5+g pool-disk-sc3-li273a DoorTransferFinished 0000B39A4147B483418ABDAF548DB36166CC] Message processing failed: Attribute is not defined: SIZE
java.lang.IllegalStateException: Attribute is not defined: SIZE
        at org.dcache.vehicles.FileAttributes.guard(FileAttributes.java:180) ~[dcache-vehicles-4.2.34.jar:4.2.34]
        at org.dcache.vehicles.FileAttributes.getSize(FileAttributes.java:351) ~[dcache-vehicles-4.2.34.jar:4.2.34]
        at diskCacheV111.services.space.SpaceManagerService.transferFinished(SpaceManagerService.java:809) [dcache-spacemanager-4.2.34.jar:4.2.34]
        at diskCacheV111.services.space.SpaceManagerService.processMessageTransactionally_aroundBody0(SpaceManagerService.java:658) [dcache-spacemanager-4.2.34.jar:4.2.34]
        at diskCacheV111.services.space.SpaceManagerService$AjcClosure1.run(SpaceManagerService.java:1) ~[dcache-spacemanager-4.2.34.jar:4.2.34]
        at org.springframework.transaction.aspectj.AbstractTransactionAspect.ajc$around$org_springframework_transaction_aspectj_AbstractTransactionAspect$1$2a73e96cproceed(AbstractTransactionAspect.aj:66) ~[spring-aspects-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.springframework.transaction.aspectj.AbstractTransactionAspect$AbstractTransactionAspect$1.proceedWithInvocation(AbstractTransactionAspect.aj:72) ~[spring-aspects-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:282) ~[spring-tx-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at org.springframework.transaction.aspectj.AbstractTransactionAspect.ajc$around$org_springframework_transaction_aspectj_AbstractTransactionAspect$1$2a73e96c(AbstractTransactionAspect.aj:70) ~[spring-aspects-4.3.8.RELEASE.jar:4.3.8.RELEASE]
        at diskCacheV111.services.space.SpaceManagerService.processMessageTransactionally(SpaceManagerService.java:648) [dcache-spacemanager-4.2.34.jar:4.2.34]
        at diskCacheV111.services.space.SpaceManagerService.processMessage(SpaceManagerService.java:597) [dcache-spacemanager-4.2.34.jar:4.2.34]
        at diskCacheV111.services.space.SpaceManagerService.access$200(SpaceManagerService.java:128) [dcache-spacemanager-4.2.34.jar:4.2.34]
        at diskCacheV111.services.space.SpaceManagerService$2.process(SpaceManagerService.java:534) [dcache-spacemanager-4.2.34.jar:4.2.34]
        at diskCacheV111.services.space.SpaceManagerService$FibonacciBackoffMessageProcessor.run(SpaceManagerService.java:1372) [dcache-spacemanager-4.2.34.jar:4.2.34]
        at org.dcache.util.CDCExecutorServiceDecorator$WrappedRunnable.run(CDCExecutorServiceDecorator.java:149) [dcache-core-4.2.34.jar:4.2.34]
        at org.dcache.util.BoundedExecutor$Worker.run(BoundedExecutor.java:251) [dcache-common-4.2.34.jar:4.2.34]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [na:1.8.0_181]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [na:1.8.0_181]
        at java.lang.Thread.run(Thread.java:748) [na:1.8.0_181]
```

This happens when a webdav transfer fails with  "Premature end of Content-Length delimited message body" or "Checksum mismatch" errors

```
02 Sep 2019 14:54:04 (pool-disk-sc3-li273a) [door:webdav-ccdcatli302@webdav-ccdcatli302Domain:AAWRkMuH5+g RemoteTransferManager PoolAcceptFile 0000B39A4147B483418ABDAF548DB36166CC] Transfer failed: org.apache.http.ConnectionClosedException: Premature end of Content-Length delimited message body (expected: 4000000000; received: 901842296
02 Sep 2019 14:54:04 (pool-disk-sc3-li273a) [door:webdav-ccdcatli302@webdav-ccdcatli302Domain:AAWRkMuH5+g RemoteTransferManager PoolAcceptFile 0000B39A4147B483418ABDAF548DB36166CC] Marking pool entry 0000B39A4147B483418ABDAF548DB36166CC on pool-disk-sc3-li273a as BROKEN
02 Sep 2019 14:54:04 (pool-disk-sc3-li273a) [door:webdav-ccdcatli302@webdav-ccdcatli302Domain:AAWRkMuH5+g RemoteTransferManager PoolAcceptFile 0000B39A4147B483418ABDAF548DB36166CC] Transfer failed in post-processing: Checksum mismatch (expected=[1:98748d49], actual=[1:485d0056])
```
dCache 4.2.34 used.

Regards,
Adrien

**Comment by jstarek** [comment]
Hi Adrien, sorry for only noticing this ticket now. While this may be a better fit for our support mailing list, let's try solving this here. One thing I don't understand about the issue is whether the failed transfers actually leave partial files on the pool in question -- the error messages you quote in your last block suggest that. Can you please try to check this? Also, what machine are we talking about and do you have admin permissions there?

**Comment by ageorget** [comment]
Hi Jürgen,
Paul asked me to open a github issue instead of a rt ticket so others can comment on this problem.

I have admin access to the machines yes, All the machines hosting DOMA test pools are concerned. I tried to find some trace of failed transfers files but I couldn't find anything relevant. The file does not appear in pool listing (rep ls) nor in the /data on the machine. 
Paul thought about a problem with cancelled HTTP-TPC transfers with space-manager enabled.

**Comment by ageorget** [comment]
Hi,
Some news about this issue. After upgrading to 5.2.9 last week, the _Attribute is not defined: SIZE_ error disappeared from the Spacemanager logs.
I don't have full DOMA pools anymore, I didn't find anything in the dCache release notes to explain why... I'll keep an eye on it and take you inform if it happen again in the future.

**Comment by paulmillar** [comment]
Thanks for the update.
No, I don't believe we did anything to dCache that would have fixed this, so it is likely because you don't have full pools.  That's a helpful hint that may allow me to reproduce the problem!

**Comment by ageorget** [comment]
Well, this happened again today with DOMA pools.
Some pools were full without any active movers and after restarting the pool, the space occupation fell from 99.9% to 28%.
```
--- rep (Repository manager) ---
State : OPEN
Files : 187
Disk space
    Total    : 2000G
    Used     : 2146512411716    [0.9995477328617126]
    Free     : 971236284    Gap : 4294967296
```

After restart : 
```
--- rep (Repository manager) ---
State : OPEN
Files : 187
Disk space
    Total    : 2000G
    Used     : 616186107702    [0.2869340161336586]
    Free     : 1531297540298    Gap : 4294967296
```

**Comment by paulmillar** [comment]
The stack-trace is reminiscent of #4311.

**Comment by DmitryLitvintsev** [comment]
I haven't noticed this ticket until now.  I recall discussing it on dcache slack back in Sep 2019. Also DOA TPC test

Something ominous.
10:18
I have not had time to investigate fully. But the following happened. We run master our TPC test stand. I have notice out of space errors on TPC monitoring. Well, I went about and just deleted all files with access time older than 30 days. To my surpsied the space has not been released. (REPLICA/ONLOINE) (edited) 
10:18
:
10:19
[stkendca06a] (rw-stkendca04a-1@rw-stkendca04a-1Domain) admin > info -a 
--- csm (Checksum module) ---
          Checksum type : adler32


--- default-transfer-service (Generic transfer service) ---
......
--- rep (Repository manager) ---
State : OPEN
Files : 13
Disk space
    Total    : 7400G
    Used     : 7941871734460    [0.9995195177031329]
    Free     : 3817763140    Gap : 4294967296
    Precious : 0    [0.0]
    Removable: 0    [0.0]
10:20
Note uses is reported as 7941871734460 (7396G).
10:20
BUT ! there are only 13 GB files! (edited) 
10:21
[stkendca06a] (rw-stkendca04a-1@rw-stkendca04a-1Domain) admin > rep ls
00006A645D89EFD7418CBB7BA798C0D90809 <C-------X--L(0)[0]> 4000000000 si={sql:stress}
000011229E534A88492D98B04BFB6C10B717 <--C--------L(0)[0]> 1994689344 si={sql:stress}
000005CD377FC82D40568E12C467FDE3940C <--C--------L(0)[0]> 3166619616 si={sql:stress}
0000F2896B96DB4E4ECEAA3FF350AA275597 <C-------X--L(0)[0]> 4000000000 si={sql:stress}
0000FA1359A0AF654F95B398A2AD87CC41E9 <--C--------L(0)[0]> 2140396384 si={sql:stress}
00001E5549E0D53147D8928F50D10BF7BB35 <--C--------L(0)[0]> 0 si={sql:stress}
0000B16839DA9BAD496CB81CDCE9AD832CAC <C-------X--L(0)[0]> 4000000000 si={sql:stress}
000052F2BDD2A975433DA35896A806BBB6B4 <--C--------L(0)[0]> 0 si={sql:stress}
0000711FD81925D14E64AE5E7524A106DC1F <--C--------L(0)[0]> 0 si={sql:stress}
00003F1EA3BEE0F44E24A66357B608AD1AF2 <C-------X--L(0)[0]> 4000000000 si={sql:stress}
00001511447EDCDE4A64AF88F236C37E5424 <C-------X--L(0)[0]> 4000000000 si={sql:stress}
000059D43F58B7B341FDA95AB9A2498B7A64 <C-------X--L(0)[0]> 4000000000 si={sql:stress}
00003F984CECEDCF49298CB335A3F1C5F30C <C-------X--L(0)[0]> 4000000000 si={sql:stress}
(edited)
10:23
total size 33G
10:23
I have restarted the pool and now
10:24
--- rep (Repository manager) ---
State : OPEN
Files : 27
Disk space
    Total    : 7400G
    Used     : 108000000000    [0.013592275413306983]
    Free     : 7837689497600    Gap : 4294967296
    Precious : 0    [0.0]
    Removable: 0    [0.0]
10:24
(some files were added as the system is constantly being tested)
10:24
So just a heads up (edited) 
10:25
We may have a problem
September 7th, 2019

dmitrylitvintsev  1:49 PM
yup ;(. Today on the empty pool (edited) 
1:53
[stkendca06a] (rw-stkendca04a-1@rw-stkendca04a-1Domain) admin > rep ls
[stkendca06a] (rw-stkendca04a-1@rw-stkendca04a-1Domain) admin > info -a
--- csm (Checksum module) ---
          Checksum type : adler32


--- default-transfer-service (Generic transfer service) ---

--- flush (Controller for centralising flushing) ---
   Flush interval                : 600000 ms
   Maximum classes flushing      : 1000
   Minimum flush delay on error  : 60000 ms
   Next flush                    : Sat Sep 07 13:57:23 CDT 2019

--- jtm (Job timeout manager) ---
  DCap (lastAccess=259200;total=432000)
  FTP (lastAccess=43200;total=432000)
  TPC (lastAccess=0;total=0)
  XRootD (lastAccess=259200;total=432000)
  HTTP (lastAccess=259200;total=432000)
  NFS (lastAccess=259200;total=432000)
  regular (lastAccess=0;total=0)
  p2p (lastAccess=43200;total=86400)

--- migration (Replica migration module client) ---

--- migration-server (Replica migration module backend) ---

--- nfs-transfer-service (NFSv4.1 transfer service) ---

--- pool (Main pool component) ---
Base directory    : /diska/rw-pool-1
Version           : 6.0.0-SNAPSHOT(a0125ba) (Sub=4)
Report remove     : ON
Pool Mode         : enabled
Clean prec. files : ON
Hsm Load Suppr.   : OFF
Ping Heartbeat    : 30 seconds
Breakeven         : 0.7
LargeFileStore    : NONE
DuplicateRequests : IGNORED
P2P File Mode     : CACHED
Mover Queue (DCap) 0(100)/0
Mover Queue (FTP) 0(30)/0
Mover Queue (TPC) 0(2)/0
Mover Queue (XRootD) 0(100)/0
Mover Queue (NFS) 0(1000)/0
Mover Queue (HTTP) 0(20)/0
Mover Queue (regular) 0(200)/0

--- post-transfer-service (Post processing execution service) ---

--- pp (Pool to pool transfer manager) ---

--- queue (HSM flush queue manager) ---
   Classes  : 0
   Requests : 0

--- remote-gsiftp-transfer-service ---

--- remote-http-transfer-service ---

--- rep (Repository manager) ---
State : OPEN
Files : 0
Disk space
    Total    : 7400G
    Used     : 439394558040    [0.05529973933322204]
    Free     : 7506294939560    Gap : 4294967296
    Precious : 0    [0.0]
    Removable: 0    [0.0]
File system
    Size : 7999865749504
    Free : 7999306846208    [0.999930135915589]
Limits for maximum disk space
    File system          : 8438701404248
    Statically configured: 9223372036854775807
    Runtime configured   : 7945689497600

--- storagehandler (Nearline storage manager) ---
 Restore Timeout  : 604800 seconds
   Store Timeout  : 604800 seconds
  Remove Timeout  : 14400 seconds
  Job Queues (active/queued)
    to store   0/0
    from store 0/0
    delete     0/0
(edited)
1:54
No files. But ued space is reported :scream:. (edited) 
September 9th, 2019
